src/widgets-service.js

```diff
@@ -150,6 +151,16 @@ WidgetsService.prototype.createTimeSeriesModel = function (attrs, layer) {
   return widgetModel;
 };
 
+WidgetsService.prototype.setWidgetsState = function (state) {
```

Comment:
also I'm guessing that the ajax request would use `dashboard.setState()`, not the function in `WidgetsService`